dragonchain/broadcast_processor/broadcast_processor.py

```diff
@@ -110,6 +120,74 @@ def make_broadcast_futures(session: aiohttp.ClientSession, block_id: str, level:
     return broadcasts
 
 
+def get_level_from_storage_location(storage_location: str):
+    result = re.search("BLOCK/.*?-l([2-5])-", storage_location)
+    if result is None:
+        return None
+    return result.group(1)
+
+
+def get_notification_urls(key: str) -> set:
+    try:
+        urls = set(VERIFICATION_NOTIFICATION[key])
+    except KeyError:
+        urls = set()
+    return urls
+
+
+def sign(message: bytes) -> str:
+    return keys.get_my_keys().make_signature(message, crypto.SupportedHashes.sha256)
+
+
+async def process_verification_notifications(session: aiohttp.ClientSession) -> None:
+    """Main function for the verification notification broadcast system
+
+    Retrieves verifications that need to be processed and sends http requests
+
+    Args:
+        session: aiohttp session for http requests
+    """
+    if VERIFICATION_NOTIFICATION:
+        futures = set()
+        for storage_location in await broadcast_functions.get_notification_verifications_for_broadcast_async():
+            verification_bytes = storage.get(storage_location)
+            level = get_level_from_storage_location(storage_location)
```

Comment:
you don't check if this ever comes back none.

This would've been caught by mypy had the return type annotation existed for the `get_level_from_storage_location` method

Reply:
true that. also typed that function. change coming up

Reply:
fixed in the `get_all_notification_endpoints` above